src/main/java/seedu/address/logic/parser/ListAppointmentCommandParser.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DOCTOR_NRIC;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PATIENT_NRIC;
+import static seedu.address.model.NewModel.PREDICATE_SHOW_ALL_APPOINTMENTS;
+
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.ListAppointmentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.Appointment;
+import seedu.address.model.appointment.AppointmentEqualDoctorNricPredicate;
+import seedu.address.model.appointment.AppointmentEqualPatientNricPredicate;
+import seedu.address.model.appointment.AppointmentFilterByNricPredicate;
+import seedu.address.model.person.Nric;
+
+/**
+ * Parses input arguments and creates a new ListAppointmentCommand object
+ */
+public class ListAppointmentCommandParser implements NewParser<ListAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListAppointmentCommand
+     * and returns an ListAppointmentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListAppointmentCommand parse(String args) throws ParseException {
```

Comment:
is there a need to pass `PREDICATE_SHOW_ALL_APPOINTMENTS` for the parameters that are not provided?